modules/service/src/main/scala/lucuma/odb/service/ObservationService.scala

```diff
@@ -44,40 +53,51 @@ object ObservationService {
   import InsertObservationResponse._
 
   def fromUserAndSession[F[_]: MonadCancelThrow](user: User, session: Session[F]): ObservationService[F] =
-    new ObservationService[F] {
-
-      def insertObservation(
-        programId:   Program.Id,
-        name:        Option[NonEmptyString],
-        existence:   Existence,
-        status:      ObsStatus,
-        activeState: ObsActiveStatus
-      ): F[InsertObservationResponse] = {
-        val af = Statements.insertObservationAs(user, programId, name, existence, status, activeState)
-        session.prepare(af.fragment.query(observation_id)).use { pq =>
-          pq.option(af.argument).map {
-            case Some(oid) => Success(oid)
-            case None      => NotAuthorized(user)
-          }
+    (programId:     Program.Id,
+     name:          Option[NonEmptyString],
+     existence:     Existence,
+     status:        ObsStatus,
+     activeState:   ObsActiveStatus,
+     constraintSet: ConstraintSet
+    ) => {
+      val af = Statements.insertObservationAs(user, programId, name, existence, status, activeState, constraintSet)
+      session.prepare(af.fragment.query(observation_id)).use { pq =>
+        pq.option(af.argument).map {
+          case Some(oid) => Success(oid)
+          case None      => NotAuthorized(user)
         }
       }
-
     }
 
   object Statements {
     import ProgramService.Statements.{ existsUserAsPi, existsUserAsCoi, existsAllocationForPartner }
 
     def insertObservationAs(
-      user: User,
-      programId:   Program.Id,
-      name:        Option[NonEmptyString],
-      existence:   Existence,
-      status:      ObsStatus,
-      activeState: ObsActiveStatus
+      user:          User,
+      programId:     Program.Id,
+      name:          Option[NonEmptyString],
+      existence:     Existence,
+      status:        ObsStatus,
+      activeState:   ObsActiveStatus,
+      constraintSet: ConstraintSet
     ): AppliedFragment = {
 
       val insert: AppliedFragment =
-        InsertObservation.apply(programId ~ name ~ existence ~ status ~ activeState)
+        InsertObservation.apply(
+          programId    ~
+           name        ~
+           existence   ~
+           status      ~
+           activeState ~
+           constraintSet.cloudExtinction ~
+           constraintSet.imageQuality    ~
+           constraintSet.skyBackground   ~
+           constraintSet.waterVapor      ~
+           ElevationRange.airMass.getOption(constraintSet.elevationRange).map(am => PosBigDecimal.unsafeFrom(am.min.value)) ~  // TODO: fix in core
```

Comment:
The GraphQL Schema has `PosBigDecimal` scalar.  Given the `AirMass.DecimalValue` you can safely get a `PosBigDecimal`.  Need to add it to the `DecimalValue` companion in core.